js/src/redux/providers/personalReducer.js

```diff
@@ -25,16 +25,16 @@ export default handleActions({
     const accounts = {};
     const contacts = {};
 
-    Object.keys(accountsInfo).forEach((address) => {
-      const account = accountsInfo[address];
-      const { name, meta, uuid } = account;
-
-      if (uuid) {
-        accounts[address] = { address, name, meta, uuid };
-      } else {
-        contacts[address] = { address, name, meta };
-      }
-    });
+    Object.keys(accountsInfo)
+      .map((address) => Object.assign({}, accountsInfo[address], { address }))
+      .filter((account) => !account.meta.deleted)
+      .forEach((account) => {
+        if (account.uuid) {
```

Comment:
```
if (account.uuid) {
          accounts[account.address] = account;
        } else {
          contacts[account.address] = account;
        }
```

One moves it to the accounts bucket, the others to the contacts bucket. (As soon as we have https://github.com/ethcore/parity/issues/2115 - there will probably be a contracts bucket as well)